mappings/level2/mast.yml

```diff
@@ -417,49 +417,211 @@ datasets:
           time:
             imas: "gas_injection.time"
 
-      # valve_voltage:
-      #   imas: "gas_injection.valve[:].voltage.data"
-      #   source: 
-      #     - name: "XDC/GAS"
-      #       shot_range:
-      #         shot_min: 28356
-      #         shot_max: 30471
-      #       channels:
-      #         - "XDC/GAS/T/G1"
-      #         - "XDC/GAS/T/G2"
-      #         - "XDC/GAS/T/G3"
-      #         - "XDC/GAS/T/G4"
-      #         - "XDC/GAS/T/G5"
-      #         - "XDC/GAS/T/G6"
-      #         - "XDC/GAS/T/G7"
-      #         - "XDC/GAS/T/G8"
-      #         - "XDC/GAS/T/G9"
-      #         - "XDC/GAS/T/G10"
-      #         - "XDC/GAS/T/G11"
-      #         - "XDC/GAS/T/G12"
-
-      #     - name: "XDC/GAS"
-      #       shot_range:
-      #         shot_min: 1
-      #         shot_max: 28355
-      #       channels:
-      #         - "XDC_GAS_T_G1"
-      #         - "XDC_GAS_T_G2"
-      #         - "XDC_GAS_T_G3"
-      #         - "XDC_GAS_T_G4"
-      #         - "XDC_GAS_T_G5"
-      #         - "XDC_GAS_T_G6"
-      #         - "XDC_GAS_T_G7"
-      #         - "XDC_GAS_T_G8"
-      #         - "XDC_GAS_T_G9"
-      #         - "XDC_GAS_T_G10"
-      #         - "XDC_GAS_T_G11"
-      #         - "XDC_GAS_T_G12"
+      valve_voltage:
+        imas: "gas_injection.valve[:].voltage.data"
+        units: "V"
+        source: 
+          - name: "XDC/GAS/"
+            shot_range:
+              shot_min: 28356
+              shot_max: 30471
+            groupings: {
```

Comment:
I think we should make this more generic. Instead of `groupings` call it `attributes` and then add `groups` as an entry within `attributes`